src/constants.ts

```diff
@@ -23,4 +23,54 @@ export const TOOLTIP_COLOR = '#3B3649';
 export const SIDE_PANEL_WIDTH: number = 280;
 export const MIN_WIDTH_TO_SHOW_SIDE_PANEL: number = 580;
 export const VIEWER_INITIAL_WIDTH: number = 500;
-export const VIEWER_HEIGHT: number = 529;
+export const VIEWER_HEIGHT: number = 580;
+
+//todo get from viewer
+export interface DisplayStateEntry {
+  name: string;
+  id: string;
+  color: string;
+}
+
+export interface UIDisplayEntry {
+  name: string;
+  displayStates: DisplayStateEntry[];
+  color: string;
+}
+
+// maps names/display states for viewer viz status (hidden/highligted, or not, in the viewport)
+export interface VisibilitySelectionMap {
+  [key: string]: string[];
+}
+
+// types for matching agent/subagent names to current status of checkboxes in the agent tree UI
+// (hidden, visible, indeterminate) - used for visibility and highlighting
+export enum ActiveState {
+  Active = 'Active',
+  Inactive = 'Inactive',
+  Indeterminate = 'Indeterminate',
+}
+
+export interface UIVisibilityMap {
+  [key: string]: ActiveState;
```

Comment:
these can never be intermediate, so I think your enum doesn't quite work here. I think you were trying to make something as general as possible but I think it makes it more confusing. How about:
```
enum HiddenState {
True 
False 
}

enum HighlightState {
True
False
}